### PR TITLE
MKT-7525

### PR DESCRIPTION
**Pull request by abhishek305**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=yw4dXkt)

Change:
- Prototype pollution-related fixes

### CODE REVIEW

File: src/entry.ts [hunk 1]
```diff
@@ -90,6 +90,24 @@ class Entry {
...
+    * (i.e., not inherited from the prototype chain) before accessing it. This helps
+    * mitigate prototype pollution vulnerabilities.
+    * 
+    * @param {GenericObjectType} obj - The object from which to retrieve the property.
+    * @param {string | number} key - The key of the property to retrieve.
+    * @returns {any} - The value of the property if it exists, otherwise undefined.
+    */
+   
+    getPropertySafely(obj: GenericObjectType, key: string | number) {
+        return Object.prototype.hasOwnProperty.call(obj, key) ? obj[key] : undefined;
```
Comment (rijil-tr) [inline src/entry.ts:108]: Is this same as that of obj. hasOwnProperty(key)? What is the deference?

File: src/field.ts [hunk 2]
```diff
@@ -87,11 +86,11 @@ class Field {
                     ? fieldObj.schema.$uid
                     : fieldObj.uid;
             const path = schemaPath.split(".");
-            let value = event.data;
-
+            let value: GenericObjectType = new Map(Object.entries(event.data));
```
Comment (rijil-tr) [inline src/field.ts:89]: Why is this needed? Instead of using GenericObjectType either declare it as indexable or something with data & indexable will solve the problem. This solution doesn't make sense.

**Review by rijil-tr — CHANGES_REQUESTED** [review]

File: src/entry.ts (hunk 1 above)
Comment (abhishek305) [reply src/entry.ts:108]: This is a utility handler. So, here no object properties are inherited, and it would take `obj`, i.e., value and `key` from forEach to work in the context and update `value` accordingly.
This is checked and working as expected

**Review by rijil-tr — APPROVED** [review]
LGTM
